### PR TITLE
IMPRO-1504: Reliability calibration table creation CLI

### PR DESCRIPTION
**Pull request by bayliffe**
A CLI for the reliabiltiy calibration table creation plugin. This is very simple. There is one acceptance test as the CLI does very little. Please see the notebook for details on the construction of the acceptance test.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/cli/construct_reliability_tables.py  1
- improver_tests/acceptance/test_construct_reliability_tables.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4942897&bid=16339718)

**Comment by bayliffe** [comment]
- Attributes added to input cubes and KGO updated.
- I definitely had a reason I decided against using units of hours and switched it to 1, but I cannot remember it and didn't write it down. As such, I've set the unit as 'hour' and if it ever comes back to me I can always change it (I doubt it).

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- improver/cli/construct_reliability_tables.py  1
- improver_tests/acceptance/test_construct_reliability_tables.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4942897&bid=16339718)

File: improver/cli/construct_reliability_tables.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    reliability tables. Reliability tables are returned as a cube with a
+    leading threshold dimension that matches that of the forecast probability
+    cubes and the thresholded truth.
+
+    Args:
+        cubes (list of iris.cube.Cube):
+            A list of cubes containing the historical probability forecasts and
+            corresponding truths used for calibration. These cubes must include
+            the same diagnostic name in their names, and must both have
+            equivalent threshold coordinates. The cubes will be distinquished
```
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver/cli/construct_reliability_tables.py]: Oops.

File: improver/calibration/reliability_calibration.py [hunk 2]
```diff
@@ -232,7 +232,7 @@ def _create_cycle_hour_coord(self, forecast_reference_time):
         """
         cycle_hour, = self._get_cycle_hours(forecast_reference_time)
         cycle_coord = iris.coords.DimCoord(cycle_hour, long_name='cycle_hour',
-                                           units=1)
+                                           units='hour')
```
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver/calibration/reliability_calibration.py]: Done.

**Pull request merged**